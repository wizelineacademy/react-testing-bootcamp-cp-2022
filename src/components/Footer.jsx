import React from "react";

function Footer() {
	return (
		<footer data-testid="footer">
			Project created during Wizeline Academy React Testing Bootcamp.
		</footer>
	);
}

export default Footer;
