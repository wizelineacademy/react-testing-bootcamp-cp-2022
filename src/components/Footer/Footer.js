import styled from "styled-components";

export const StyledFooter = styled.nav`
  color: white;
  background-color: blue;
  height: 4rem;
  text-align: center;
  padding-top: 20px;
  font-size: 1.8rem;
  box-shadow: 1px 1px 3px 0.5px darkgray;
`;

const Footer = () => {
  return (
    <StyledFooter>
      Project created during Wizeline Academy React Testing Bootcamp
    </StyledFooter>
  );
};

export default Footer;
