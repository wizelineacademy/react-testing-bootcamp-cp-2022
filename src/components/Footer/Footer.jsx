import styled from "styled-components";

const Footer = () => (
	<footer>
		<FooterParagraph>
			Project created during Wizeline Academy React Testing Bootcamp
		</FooterParagraph>
	</footer>
);

const FooterParagraph = styled.p`
	text-align: center;
`;

export default Footer;
