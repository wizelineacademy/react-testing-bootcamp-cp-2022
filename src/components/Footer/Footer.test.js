import { render, screen } from "@testing-library/react";

import Footer from ".";

const setUp = () => render(<Footer />);

describe("Footer", () => {
	it("should render the footer text", () => {
		const { container } = setUp();

		const footer = screen.getByText(
			/project created during wizeline academy react testing bootcamp/i
		);

		expect(footer).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
