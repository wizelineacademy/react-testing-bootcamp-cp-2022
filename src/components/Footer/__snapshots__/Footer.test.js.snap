// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render the footer text 1`] = `
<div>
  <footer>
    <p
      class="sc-bczRLJ hAGsLM"
    >
      Project created during Wizeline Academy React Testing Bootcamp
    </p>
  </footer>
</div>
`;
